lib/axon.ex

```diff
@@ -3495,8 +3487,36 @@ defmodule Axon do
   or backend.
   """
   @doc type: :compilation
-  def compile(model, opts \\ []) when is_list(opts) do
-    {Axon.Compiler.compile_init(model, opts), Axon.Compiler.compile_predict(model, opts)}
+  def build(model, opts \\ []) when is_list(opts) do
+    {init_fn, predict_fn} = Axon.Compiler.build(model, opts)
+    opts = [on_conflict: :reuse] ++ opts
+    # TODO: Wrap in jit_apply
+    {&Nx.Defn.jit(init_fn, [&1, &2], opts), &Nx.Defn.jit(predict_fn, [&1, &2], opts)}
```

Comment:
And you either want to move the Nx.Defn.jit call to the Compiler... or keep it here but then `Keyword.pop(opts, :mode)` here.